### DISCUSSION TITLE
Option to disable the hyperlink tooltip/status bar

### DISCUSSION THREAD

**Discussion by wzxu** [ghostty-org/ghostty#2654] [Feature Requests, Ideas] [answered]
Would like an option to never show this:
I find it pretty redundant as I can already see the working directory/filename or URL from the text that makes the link.
![Image](https://github.com/user-attachments/assets/c21141b7-e774-45b4-8bd0-adb7adb0b3dd)

(To be honest I'd maybe tolerate it if it at least looks like iTerm's, it's just pretty ugly as it is currently.)
![Image](https://github.com/user-attachments/assets/f37cae2b-c6a8-45cd-9690-809a427952a6)
(The secure input indication is BEAUTIFUL, though. Gotta show my appreciation 💯)

**Comment by mitchellh** [comment]
Issues are not the way to make a feature request. Please see [CONTRIBUTING.md](https://github.com/ghostty-org/ghostty/blob/main/CONTRIBUTING.md). I've converted this to a discussion.

**Comment by mitchellh** [comment, marked as answer]
I'm not sure I like the idea of making this configurable. While you can easily see the URL that you _think_ you're going to click, it could just as easily be an OSC8 link that goes somewhere else. For example, using HTML as pseudocode: `<a href="http://bad.example.com">http://google.com</a>`. The equivalent is possible with OSC8.

I think conditionally only showing the tooltip when it's OSC8 is more realistic but I'm just not sure if I like that level of configurability.

**Reply by wzxu** [reply]
Hmm if that's the case (wasn't aware HTML links also work) then yeah a hint may be a must.

I guess this indeed depends on how each terminal project sees the topic and not quite user-configurable,
seeing that some terminals, like kitty, WezTerm, Alacritty, just downright don't show them and also not configurable AFAIK.

**Reply by mitchellh** [reply]
Yes, not its not HTML links exactly, it's OSC8, but semantically they're the same.

**Comment by mitchellh** [comment]
I'm going to close this discussion as not planned due to the given reasons in my prior comment.

**Reply by wzxu** [reply]
Would it be possible to adjust its appearance then? 
Rounded corners and away from the edge like Safari's will do just fine.
![CleanShot 2024-11-15 at 11 35 22@2x](https://github.com/user-attachments/assets/6c624c50-f7d4-4b52-a323-48ffd62954a8)
(Safari doesn't do so good with the radius here though…)

**Reply by jparise** [reply]
> Would it be possible to adjust its appearance then? Rounded corners and away from the edge like Safari's will do just fine.

I'm trying out a version of this in #2684.

**Comment by mkrieger1** [comment]
While I appreciate the reasoning for having the tooltip, more often than it helps me open a link, it obscures my prompt (or whatever is at the bottom of the terminal) because some text that looks like a link has moved under the mouse pointer and I press Ctrl as part of some key combination (see also #3792).

So I support this proposed option to disable the tooltip, or an alternative solution, like
- an option to configure the link-clicking key to be something else than Ctrl
- an option to configure the position of the tooltip
- to only show the tooltip if the mouse was actively moved on a link, not when the link has moved under the mouse (there is already a feature that hides the mouse pointer when typing, so it would seem feasible to me to hide the tooltip as well while in "mouse pointer is hidden" state).

(I found a workaround for my case, which is to set `link-url` to `false`, i.e. disable clicking on links completely.)

**Reply by 00-kat** [reply]
> an option to configure the link-clicking key to be something else than Ctrl

FYI that's #3848.

**Comment by curio77** [comment]
I find it irritating, too.  Further options:
- add an option disabling link highlighting/clicking entirely
- add an option or generally make it such that the link activation and tooltip only happen once the mouse is moved with Ctrl being pressed

**Comment by selckin** [comment]
I would also like a feature to hide the tooltip, and believe its a bit overly paranoid, the browser is still sandboxed and don't think people will try to read the tooltip to double check if its correct anyway

**Reply by jcollie** [reply]
The link may or may not be launched in a sandboxed browser, we're just running `open` (on macOS) or `xdg-open` (on GTK) so you can't guarantee what program opens the link.

As far as reading the tooltip, I do that _all the time_. I doubt that I'm the only one.

All that really has nothing to do with having an option to disable the tooltip though.

**Reply by selckin** [reply]
Feel like a warning in the docs should be enough, can't protect users from everything
I'm never clicking a link from an untrusted source

An option to disable OSC8 links would offer more protection

**Reply by jcollie** [reply]
Docs? Who reads docs? Anyway, the subject of user behavior with respect to security is really irrelevant here. There are a lot of people that _like_ the URL overlays (including many of the developers) irrespective of the security benefit. So it's going to remain on by default.

Eventually, someone _may_ get around to adding an option to disable the overlays for people that don't want it. But today is not that day. And pestering us with vague "user security behavior" anecdotes isn't helping anything.
